server/test/functions.go

```diff
@@ -0,0 +1,15 @@
+package test
+
+import (
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+// FunctionName returns the name of the function provided as the argument.
+// Behavior is undefined if a non-function is passed.
+func FunctionName(f interface{}) string {
```

Comment:
This is actually code that appeared copy-pasted in three different packages. I moved it to the test package rather than copying it into a fourth. It's useful because it allows us to print the name of the test function being executed when running tests.